src/pybind/mgr/cephadm/module.py

```diff
@@ -1810,11 +1824,16 @@ def _check_daemons(self):
                 continue
 
             # dependencies?
-            if dd.daemon_type == 'grafana':
-                # put running instances at the front of the list
-                grafanas.insert(0, dd)
-            elif dd.daemon_type == 'iscsi':
-                iscsi_daemons.append(dd)
+            if dd.daemon_type in ['grafana', 'iscsi', 'prometheus', 'alertmanager']:
+                cephadm_service = self.cephadm_services.get(dd.daemon_type)
```

Comment:
what about making self.cephadm_services a method? That would prevent others from running into the same issue?

Reply:
Yeah, thanks! I definitely prefer that approach: encapsulating service functionality within the service class scope, rather than having the consumers of the dummy classes implementing the functionality.

Regarding the use of combinations of nested dicts + lists for storing structured data, it feels to me like bringing JSON back to Python, when even Javascript is moving towards Types & Classes (ES6, Typescript, ...). It's great to use mypy, but classes provide the only runtime type checking.

Reply:
A method is created to get services. But I think all service usage in module.py should also be refactored. It will become a bit lengthy.

e.g.
```
return self._add_daemon('mon', spec, self.mon_service.create)
```
to
```
# maybe change 'mon' to a constant.
return self._add_daemon('mon', spec, self._get_cephadm_service('mon').create)
```